### PR TITLE
Add typings for [node-resque] library.

### PR DESCRIPTION
**Pull request by gordey4doronin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Compile and run screenshot 👇 
screen shot 2018-09-07 at 00 27 17

### CODE REVIEW

**Comment by gordey4doronin** [comment]
@evantahler Added typings for your library. You probably can have a brief look if you're interested. 🙂 

@minestarks Can I ask you for a review please? 😊

**Comment by evantahler** [comment]
Thanks for all this work @gordey4doronin!

I'm not a typescript user, and I'm new to this project.  Can you share why you opted to create a separate typeded here and not add annotations or jsdoc to the project itself?

**Comment by gordey4doronin** [comment]
@evantahler Sorry for not giving an answer for a month.. 😓 

* I added a typings to this project because it's just a usual approach for adding typings to any of JS projects by not-owners. Usually it's much faster to create a typings here rather than add them to original project repo.
* Also this repo has a set of nice rules to validate typings properly.
* If you wish we can move the typings to the original project repo. And deprecate them there.
* Usually typings are supported by a open-source community. Once you move them to your project - usually it's expected that they will be updated with the project source code correspondingly (and that actually feels (that is not actually) like a project maintainer responsibility rather than community.
* Not sure what you mean by annotations?
* Regarding JSDoc - I don't know if there is a tooling for generating TS typings from JSDoc. Even if such tool exists - it could be even more painful to maintain JSDoc in a format convertible to TS definitions.
* BTW I also don't see any JSDoc in the original source code. So adding them from a scratch would be longer.

TL;DR

My initial idea was to improve your JS code to being TS actually. The code is pretty clean and readable and it feels like it would be a straightforward work to do that. But next my thought was that you probably wouldn't like having your project in TS rather than JS. And maintaining an additional copy of codebase but rewritten in TS not worth it. So I ended up adding typings to that project.

**Comment by gordey4doronin** [comment]
TIL https://github.com/Microsoft/TypeScript/wiki/JSDoc-support-in-JavaScript

If you mean this? 🤔  That could work, but as I said - there is no JSDoc comments as of now. So adding typings was faster. But JSDoc is probably easier to maintain.
